test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/Common/HttpClientResponseMessage.cs

```diff
@@ -0,0 +1,134 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="HttpClientResponseMessage.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.E2E.TestCommon.Common
+{
+    /// <summary>
+    /// An implementation of <see cref="IODataResponseMessageAsync"/> that uses an <see cref="HttpResponseMessage"/> under the covers.
+    /// In ODataLibrary, a message is an abstraction which consists of stream and header interfaces that hides the details of stream-reading/writing.
+    /// </summary>
+    public class HttpClientResponseMessage : IODataResponseMessageAsync, IServiceCollectionProvider, IDisposable, IAsyncDisposable

```

Comment:
This class doesn't own the `HttpResponseMessage` object that it's disposing. The object is passed into it's constructor. The code that instantiates `HttpClientResponseMessage` should be responsible for disposing the `HttpResponseMessage` that it passes into the constructor. For that reason, I'm not sure it's necessary for this class to implement `IDisposable` and `IAsyncDisposable`.